runtime/queries/metricsview.go

```diff
@@ -292,6 +292,327 @@ func buildFilterClauseForCondition(mv *runtimev1.MetricsViewSpec, cond *runtimev
 	return fmt.Sprintf("AND (%s) ", condsClause), args, nil
 }
 
+type identifier struct {
+	mapped string
+	unnest bool
+}
+
+func newIdentifier(name string) identifier {
+	return identifier{safeName(name), false}
+}
+
+func dimensionAliases(mv *runtimev1.MetricsViewSpec) map[string]identifier {
+	aliases := map[string]identifier{}
+	for _, dim := range mv.Dimensions {
+		aliases[dim.Name] = identifier{safeName(metricsViewDimensionColumn(dim)), dim.Unnest}
+	}
+	return aliases
+}
+
+func buildFromExpression(expr *runtimev1.Expression, allowedIdentifiers map[string]identifier, dialect drivers.Dialect) (string, []any, error) {
```

Comment:
`buildFromExpression` sounds to me like "build `FROM` expression" – consider renaming to `buildExpression` or something like that